buildkitd/dockerd-wrapper.sh

```diff
@@ -280,9 +284,25 @@ load_registry_images() {
                     ;;
             esac
             echo "Pulling $with_reg and retagging as $user_tag"
-            (docker pull "$with_reg" && docker tag "$with_reg" "$user_tag" && docker rmi "$with_reg") || (stop_dockerd; exit 1)
+            # Download and tag images in parallel
+            (docker pull -q "$with_reg" && docker tag "$with_reg" "$user_tag" && docker rmi "$with_reg") &
+
+            bg_processes="$bg_processes $!"
+
         done
-        echo "...done"
+
+        # Wait for all background processes to finish
+        for pid in $bg_processes; do
+            wait "$pid" || {
+                echo "Downloading of images failed"
+                stop_dockerd
+                exit 1
+            }
+        done
+        end_time=$(date +%s%N | cut -b1-13)
```

Comment:
ultra nit: What happens after 2286-11-20 17:46:39 :laughing:

answer: not our problem.

Reply:
9999999999 = 2286-11-20 17:46:39